### PR TITLE
Fixed input behavior

### PR DESCRIPTION
**Pull request by Mav-Ivan**
https://github.com/ita-social-projects/SpaceToStudy-Client/assets/110425368/1347fbbb-ad04-45ff-a441-7087d6840c0c

### CODE REVIEW

**Review by maarekua — APPROVED** [review]

File: src/containers/offer-page/faq-block/FaqBlock.tsx [hunk 1]
```diff
@@ -54,7 +54,7 @@ const FaqBlock = <T extends CreateOrUpdateOfferData>({
   }
 
   const questionsAnswers = data.FAQ.map((el, idx, array) => (
-    <Box key={el.question} sx={styles.faqInputsBlock}>
+    <Box key={idx} sx={styles.faqInputsBlock}>
```
Comment (abalanovsky) [inline src/containers/offer-page/faq-block/FaqBlock.tsx:57]: There is sonar issue

_Do not use Array index in keys_

Comment (Mav-Ivan) [reply src/containers/offer-page/faq-block/FaqBlock.tsx]: done

**Review by Tolik170 — APPROVED** [review]

File: src/containers/offer-page/create-or-edit-offer/CreateOrEditOffer.constants.ts [hunk 2]
```diff
@@ -10,7 +10,7 @@ export const getInitialValues = (offer: Offer | null) => ({
   description: offer?.description ?? '',
   price: offer?.price.toString() ?? '',
   status: offer?.status ?? StatusEnum.Active,
-  FAQ: offer?.FAQ ?? [{ question: '', answer: '' }]
+  FAQ: offer?.FAQ ?? [{ question: '', answer: '', _id: `${Date.now()}` }]
```
Comment (dmtrth25) [inline src/containers/offer-page/create-or-edit-offer/CreateOrEditOffer.constants.ts:13]: Look. `el` item already have `_id` property, so we can we may not create another `_id` here

image

File: src/containers/offer-page/faq-block/FaqBlock.tsx [hunk 3]
```diff
@@ -54,7 +57,7 @@ const FaqBlock = <T extends CreateOrUpdateOfferData>({
   }
 
   const questionsAnswers = data.FAQ.map((el, idx, array) => (
-    <Box key={idx} sx={styles.faqInputsBlock}>
+    <Box key={el._id} sx={styles.faqInputsBlock}>
```
Comment (dmtrth25) [inline src/containers/offer-page/faq-block/FaqBlock.tsx:60]: Also can check interface for FAQ there is a `_id` property
image

image

It's only for your information.
Your approach also make sense, but as you can see it make create without adding `_id: ${Date.now()} }`


https://github.com/ita-social-projects/SpaceToStudy-Client/assets/56305508/651a24cf-cbdd-4eea-96a4-21a344324fec

**Review by dmtrth25 — APPROVED** [review]

**Pull request merged**